### ISSUE TITLE
Add option to turn off header insertion for clangd or option to pass flags to language server

### ISSUE THREAD

**Issue by pfirsich** [zed-industries/zed#20534]
### Check for existing issues

- [X] Completed

### Describe the feature

I found https://github.com/zed-industries/zed/issues/14411, which describes the problem, but the fix does not apply generally (at least not to me).

clangd will include headers when accepting code completions (--header-insertion is "iwyu" by default), which in my case breaks code whenever I use code completions.

Example:
Before accepting completion:
![Before Completion](https://github.com/user-attachments/assets/c271bc9e-8d57-4176-8240-76ca9ea6eb09)
After accepting completion:
![After Completion](https://github.com/user-attachments/assets/823389e6-89d5-4c55-a7f3-88188f848d53)

The annoying thing with this feature is that it will include headers, that might break the code. In the example above you see that `unicodeobject.h` is included, which is not supposed to be included by itself (only through `Python.h`) and especially not before `Python.h`, because `pyconfig.h` has to be included first. So whenever I accept a completion, my code breaks (everything turns red, auto-completions don't work anymore and of course it doesn't compile anymore) and I have to scroll back and remove the include.

I tried putting this .clangd in my workspace root and my home directory, but it did not disable the feature:
```
CompileFlags:
  Add: [ -header-insertion=never ]
``` 

I am just trying out Zed and I think it's very cool, but for this project I cannot use it because of this simple reason.

I think it would be great to add a configuration option to pass flags to the language server (`language_servers` might do this, but I can't find any documentation) or a flag to disable this feature directly.

### Environment

Zed: v0.159.7 (Zed)
OS: Linux Wayland debian 12
Memory: 31 GiB
Architecture: x86_64
GPU: Intel(R) Graphics (RPL-P) || Intel open-source Mesa driver || Mesa 22.3.6

clangd --version:
Debian clangd version 14.0.6
Features: linux+grpc
Platform: x86_64-pc-linux-gnu

**Comment by Poldraunic** [comment]
Agreed. This type of behaviour is especially painful when working with Qt. It doesn't break the code, like in your case, but it pull "private" header files (such as `#include "qstring.h"`) instead of canonical (`#include <QString>`).

It took a while but eventually I've found this kind soul whose suggestion did help. https://github.com/zed-industries/zed/discussions/6629#discussioncomment-10493418. This configuration is not documented anywhere (at least not [here](https://zed.dev/docs/configuring-zed#lsp) nor [here](https://zed.dev/docs/configuring-languages#using-language-server-features)). How this person figured it out I will never know.

But I will say there should be something in the docs on how to set up "unsupported" language server.

**Comment by hokein** [comment]
Two ways to disable header-insertion behavior in clangd:

1. via the `.clangd` config ([recommended](https://clangd.llvm.org/config#headerinsertion)):

```
Completion:
  HeaderInsertion: Never
```

2. via the "clangd" command-line option:
 ```json
  "lsp": {
    "clangd": {
      "binary": {
        "arguments": ["--header-insertion=never"]
      }
    }
  }
```
